http-server/src/server.rs

```diff
@@ -224,16 +225,16 @@ impl Server {
 						// Our [issue](https://github.com/paritytech/jsonrpsee/issues/296).
 						if let Ok(req) = serde_json::from_slice::<JsonRpcRequest>(&body) {
 							// NOTE: we don't need to track connection id on HTTP, so using hardcoded 0 here.
-							methods.execute(&tx, req, 0).await;
+							if let Some(fut) = methods.execute(&tx, req, 0) {
+								fut.await;
+							}
 						} else if let Ok(_req) = serde_json::from_slice::<JsonRpcNotification<Option<&RawValue>>>(&body)
 						{
 							return Ok::<_, HyperError>(response::ok_response("".into()));
 						} else if let Ok(batch) = serde_json::from_slice::<Vec<JsonRpcRequest>>(&body) {
 							if !batch.is_empty() {
 								single = false;
-								for req in batch {
-									methods.execute(&tx, req, 0).await;
-								}
+								join_all(batch.into_iter().filter_map(|req| methods.execute(&tx, req, 0))).await;
```

Comment:
Interesting, so here you don't need to attach the `.await` to each executed method but instead can attach it to the `join_all`, and it will treat each request as an async method? Very cool.

Reply:
It's basically the equivalent of `Promise.all` :)

Reply:
Ahhh, awesome thanks. That makes a ton of sense :)